contracts/shapella-upgrade/ShapellaUpgradeTemplate.sol

```diff
@@ -407,67 +377,106 @@ contract ShapellaUpgradeTemplate {
-    string public constant VALIDATOR_DELINQUENT_TIMEOUT_IN_SLOTS_KEY = "VALIDATOR_DELINQUENT_TIMEOUT_IN_SLOTS";
-    bytes public constant VALIDATOR_DELINQUENT_TIMEOUT_IN_SLOTS_VALUE = hex"7080";
-    string public constant PREDICTION_DURATION_IN_SLOTS_KEY = "PREDICTION_DURATION_IN_SLOTS";
-    bytes public constant PREDICTION_DURATION_IN_SLOTS_VALUE = hex"c4e0";
-    string public constant FINALIZATION_MAX_NEGATIVE_REBASE_EPOCH_SHIFT_KEY = "FINALIZATION_MAX_NEGATIVE_REBASE_EPOCH_SHIFT";
-    bytes public constant FINALIZATION_MAX_NEGATIVE_REBASE_EPOCH_SHIFT_VALUE = hex"0546";
-    string public constant NODE_OPERATOR_NETWORK_PENETRATION_THRESHOLD_BP_KEY = "NODE_OPERATOR_NETWORK_PENETRATION_THRESHOLD_BP";
-    bytes public constant NODE_OPERATOR_NETWORK_PENETRATION_THRESHOLD_BP_VALUE = hex"64";
+    //
+    // Values for checks to compare with or other
+    //
+    uint256 internal constant HC_FOR_AO_EPOCHS_PER_FRAME = 225;
+    uint256 internal constant HC_FOR_VEBO_EPOCHS_PER_FRAME = 75;
+    uint256 internal constant HC_FAST_LANE_LENGTH_SLOTS = 10;
+    // HC_FAR_FUTURE_EPOCH calculated by `_computeEpochAtTimestamp(type(uint64).max)` as in HashConsensus
+    uint256 internal constant HC_FAR_FUTURE_EPOCH = 48038396021100853;
+
+    uint256 internal constant EXPECTED_FINAL_LIDO_VERSION = 2;
+    uint256 internal constant EXPECTED_FINAL_NODE_OPERATORS_REGISTRY_VERSION = 2;
+    uint256 internal constant EXPECTED_FINAL_LEGACY_ORACLE_VERSION = 4;
+    uint256 internal constant EXPECTED_FINAL_ACCOUNTING_ORACLE_VERSION = 1;
+    uint256 internal constant EXPECTED_FINAL_STAKING_ROUTER_VERSION = 1;
+    uint256 internal constant EXPECTED_FINAL_VALIDATORS_EXIT_BUS_ORACLE_VERSION = 1;
+    uint256 internal constant EXPECTED_FINAL_WITHDRAWAL_QUEUE_VERSION = 1;
+    uint256 internal constant EXPECTED_FINAL_WITHDRAWAL_VAULT_VERSION = 1;
+
+    uint256 internal constant EXPECTED_DSM_MAX_DEPOSITS_PER_BLOCK = 150;
+    uint256 internal constant EXPECTED_DSM_MIN_DEPOSIT_BLOCK_DISTANCE = 25;
+    uint256 internal constant EXPECTED_DSM_PAUSE_INTENT_VALIDITY_PERIOD_BLOCKS = 6646;
+
+    uint256 internal constant SANITY_LIMIT_churnValidatorsPerDayLimit = 40000;
+    uint256 internal constant SANITY_LIMIT_oneOffCLBalanceDecreaseBPLimit = 500;
+    uint256 internal constant SANITY_LIMIT_annualBalanceIncreaseBPLimit = 1000;
+    uint256 internal constant SANITY_LIMIT_simulatedShareRateDeviationBPLimit = 50;
+    uint256 internal constant SANITY_LIMIT_maxValidatorExitRequestsPerReport = 500;
+    uint256 internal constant SANITY_LIMIT_maxAccountingExtraDataListItemsCount = 500;
+    uint256 internal constant SANITY_LIMIT_maxNodeOperatorsPerExtraDataItemCount = 100;
+    uint256 internal constant SANITY_LIMIT_requestTimestampMargin = 7680;
+    uint256 internal constant SANITY_LIMIT_maxPositiveTokenRebase = 750000;
+
+    string internal constant NORMALIZED_CL_REWARD_PER_EPOCH_KEY = "NORMALIZED_CL_REWARD_PER_EPOCH";
```

Comment:
Not sure, but can the `string` items be `bytes` here?

Reply:
Seems, it doesn't help with bytecode.
```solidity
    string internal constant FOO_STRING = "NORMALIZED_CL_REWARD_PER_EPOCH";
    ...
    revert Debug(FOO_STRING);
```
and
```solidity
    bytes internal constant FOO_BYTES = "NORMALIZED_CL_REWARD_PER_EPOCH";
    ...
    revert Debug(FOO_BYTES);
```
result in the same bytecode size.